presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -1563,6 +1563,47 @@ public void testKeyBasedSamplingInlined()
         assertQuerySucceeds(session, "select count(1) from orders join lineitem using(orderkey)");
     }
 
+    @Test
+    public void testColumnFilter()
+    {
+        String tmpTableName = generateRandomTableName();
+        assertUpdate(format("CREATE TABLE %s " +
+                "AS " +
+                "SELECT c_boolean, c_bigint, c_double, c_timestamp, c_varchar, c_varbinary " +
+                "FROM ( " +
+                "  VALUES " +
+                "    (null, null, null, null, null, null), " +
+                "    (true, BIGINT '1', DOUBLE '2.2', TIMESTAMP '2012-08-08 01:00', CAST('abc1' AS VARCHAR), to_ieee754_64(1))," +
+                "    (false, BIGINT '0', DOUBLE '1.2', TIMESTAMP '2012-08-08 00:00', CAST('abc2' AS VARCHAR), to_ieee754_64(2))," +
+                "    (true, BIGINT '2', DOUBLE '3.3', TIMESTAMP '2012-09-09 01:00', CAST('cba1' AS VARCHAR), to_ieee754_64(3)), " +
+                "    (false, BIGINT '1', DOUBLE '2.3', TIMESTAMP '2012-09-09 00:00', CAST('cba2' AS VARCHAR), to_ieee754_64(4)) " +
+                ") AS x (c_boolean, c_bigint, c_double, c_timestamp, c_varchar, c_varbinary)", tmpTableName), 5);
+
+        // NOTE: The query below does not list the c_timestamp field because Velox uses the America/Los_Angeles
```

Comment:
Ah... Do you want to use the assertQuery with a VALUES clause for verification for just this case only then ?

Reply:
@aditi-pandit Do you mean to add code like the following: 
```
assertQuery(format("SELECT c_timestamp FROM %s WHERE c_boolean", tmpTableName), "VALUES(XXX)");

assertQuery(format("SELECT c_timestamp FROM %s WHERE c_bigint = 0", tmpTableName), "VALUES(XXX)");

assertQuery(format("SELECT c_timestamp FROM %s WHERE c_double = 1.2", tmpTableName), "VALUES(XXX)");

assertQuery(format("SELECT c_timestamp FROM %s WHERE c_TIMESTAMP = TIMESTAMP '2012-09-09 00:00'", tmpTableName), "VALUES(XXX)");

assertQuery(format("SELECT c_timestamp FROM %s WHERE c_varchar = CAST('cba2' AS VARCHAR)", tmpTableName), "VALUES(XXX)");

assertQuery(session, format("SELECT c_timestamp FROM %s WHERE c_varbinary = to_ieee754_64(1)", tmpTableName), "VALUES(XXX)");
```

Reply:
@wypb : Nit. You can remove this comment.